### ISSUE TITLE
for adding predicted features: add a flag for exists or not

### ISSUE THREAD

**Issue by ClimbsRocks** [ClimbsRocks/auto_ml#139]
inside transform for AddPredictedFeature, check to see if we already have this feature present in the dataset. if we do, don't do anything! If we don't, add it in. 

and maybe add it in every time for a single row, just to be sure we're doing that right at prediction time. 

Doing this will let us get predicted features once for our training set, and once for our test set, rather than having to re-compute them each and every new model we train up.

**Comment by ClimbsRocks** [comment]
way too much complexity. removing.
